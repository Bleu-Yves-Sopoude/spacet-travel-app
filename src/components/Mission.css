  .table-container {
    max-width: 80%;
    margin: auto;
    margin-top: 50px;
    margin-bottom: 50px;
  }
  
  .table-container table {
    margin: 0 auto;
  }
  
  .table-container .working {
    vertical-align: middle;
  }
  
  .working .Bag {
    font-size: 17px;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
  }
