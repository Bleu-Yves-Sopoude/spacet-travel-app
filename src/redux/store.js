import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missionsSlice';
import rocketsReducer from './rockets/rocketsSlice.js'

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets:rocketsReducer,
  },
});

export default store;
