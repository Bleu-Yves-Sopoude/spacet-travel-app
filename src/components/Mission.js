import React, { useEffect } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import './Mission.css';
import { useDispatch, useSelector } from 'react-redux';
import { Table } from 'react-bootstrap';
import Button from 'react-bootstrap/Button';
import Badge from 'react-bootstrap/Badge';
import { fetchmissions, joinmission } from '../redux/missions/missionsSlice';

const Mission = () => {
  const dispatch = useDispatch();
  const { missions, isLoading, error } = useSelector((state) => state.missions);
  useEffect(() => {
    if (missions.length === 0) {
       dispatch(fetchmissions());
    } },
    []);

  const handlejoinmission = (missionId, reserved) => {
    if (reserved) {
      dispatch(joinmission({ missionId, reserved: false }));
    } else {
      dispatch(joinmission({ missionId, reserved: true }));
    }
  };

  if (isLoading) {
    return <p>Loading...</p>;
  }

  if (error) {
    return (
      <p>
        Error:
        {error}
      </p>
    );
  }

  return (
    <div className="table-container">
      <Table striped bordered hover size="sm">
        <thead>
          <tr>
            <th style={{ width: '180px' }}>Mission</th>
            <th>Description</th>
            <th style={{ width: '150px' }}>Status</th>
            <th style={{ width: '180px' }}>&nbsp;</th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <tr key={mission.mission_id}>
              <td><h5>{mission.mission_name}</h5></td>
              <td className="working"><p>{mission.description}</p></td>
              <td className="working">
                {mission.reserved ? (
                  <h5 className="Bag"><Badge bg="primary">Active Member</Badge></h5>
                ) : (
                  <h5 className="Bag"><Badge bg="secondary">NOT A MEMBER</Badge></h5>
                )}
              </td>
              <td className="working">
                <div className="d-flex align-items-center justify-content-center btn-container">
                  <Button variant={mission.reserved ? 'outline-danger' : 'outline-primary'} onClick={() => handlejoinmission(mission.mission_id, mission.reserved)}>
                    {mission.reserved ? 'Leave Mission' : 'Join Mission'}
                  </Button>
                </div>
              </td>
            </tr>
          ))}
        </tbody>
      </Table>
    </div>
  );
};

export default Mission;
