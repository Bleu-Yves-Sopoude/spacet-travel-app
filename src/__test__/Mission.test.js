import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import Mission from '../components/Mission';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);
describe('MyMissions', () => {
  it('render Rockets Components', () => {
    render(<Mission />);
    expect(Mission).toMatchSnapshot();
  });
});
