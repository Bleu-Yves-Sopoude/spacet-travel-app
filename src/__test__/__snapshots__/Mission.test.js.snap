// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyMissions render Mission Components 1`] = `[Function]`;
